### ISSUE TITLE
Does not work with heregexes and comments

### ISSUE THREAD

**Issue by netei** [decaffeinate/decaffeinate#86]
Input : 

```
regex= ///
[^a-z]+  #foo
[a-z]+
///
```

Output: 

```
~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:813
    throw new Error('BUG! Could not fix range for ' + node.type + ' at line '
          ^
Error: BUG! Could not fix range for Program at line 1, column 1
    at fixRange (~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:813:11)
    at ~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:723:5
    at traverse (~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:473:23)
    at parse (~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:721:3)
    at convert (~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:4861:13)
    at ReadStream.<anonymous> (~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:4816:16)
    at ReadStream.emit (events.js:129:20)
    at _stream_readable.js:908:16
    at process._tickCallback (node.js:355:11)
```

**Comment by netei** [comment]
Doesn't work with the following herexegp either : 

```
"foo".replace ///
foo
///g, "bar"
```

**Comment by eventualbuddha** [comment]
This appears to be two different issues. The heregex with the comment fails due to a parsing bug with CoffeeScriptRedux. Namely, it seems to strip the comment and treat it as if that part of the string were never there:

```js
{ type: 'Program',
  line: 1,
  column: 1,
  raw: 'regex= ///\n[^a-z]+  \n[a-z]+\n///',
  range: [ 0, 31 ],
  body: 
   { type: 'Block',
     line: 1,
     column: 1,
     raw: 'regex= ///\n[^a-z]+  \n[a-z]+\n///',
     range: [ 0, 31 ],
     statements: [ [Object] ] } }
```

That makes the `range` values for the nodes incorrect, which leads to the error you encountered.

The second heregex fails because of a bug in decaffeinate, specifically `trimmedNodeRange`. Here's what it tries to do to transform the original string:

```
INSERT 13 "("
OVERWRITE 14 26 "///\nfoo\n///g" -> "/foo/g"
INSERT 33 "\n)"
[ 0, 33 ] [ 0, 25 ]
INSERT 25 ";"
```

The `[ 0, 33 ]` is the original range of the `FunctionApplication`, and `[ 0, 25 ]` is the "trimmed" range, which is supposed to have removed whitespace and comments from the left and right of the node. Something is wrong with it, which I should have a look at this week.

**Comment by eventualbuddha** [comment]
The second issue has been fixed in v1.40.18.

**Comment by eventualbuddha** [comment]
This is now fixed, but the comments aren't preserved ([repl](http://decaffeinate.github.io/decaffeinate/repl/#?evaluate=false&code=regex%3D%20%2F%2F%2F%0A%5B%5Ea-z%5D%2B%20%20%23foo%0A%5Ba-z%5D%2B%0A%2F%2F%2F)).
